### DISCUSSION TITLE
Add different labels for different `customManagers`

### DISCUSSION THREAD

**Discussion by basil** [renovatebot/renovate#25657] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

N/A

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

My Renovate config is here: https://github.com/jenkinsci/bom/blob/master/.github/renovate.json

I want the PRs from the first of the `customManagers` (formerly `regexManagers`) to have the labels `dependencies` and `full-test`. I want the PRs from the second of the `customManagers` (formerly `regexManagers`) to have the labels `dependencies` and `weekly-test`.

### What you tried so far.

I haven't tried anything, because I can't find any examples of labels with `customManagers` in the documentation. I read all the documentation and could not find one example of this.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
@secustor @RahulGautamSingh I think we talked in the past about allowing a "name" field per custom manager? If so then we could support matchManagers=custom.regex.foo to achieve this requested feature

**Reply by secustor** [reply]
Yes we did, tough I'm not sure if I like the `custom.regex.name` construct.

**Reply by rarkins** [reply]
Any alternative?

**Reply by secustor** [reply]
I could see `custom.regex:name` or `custom:name` for example

**Reply by viceice** [reply]
I would prefer same semantics like our custom datasources

**Comment by RahulGautamSingh** [comment, marked as answer]
Since you are using the custom managers to target a single dependency and also using `depNameTemplate` to define a name for the extracted dep, you can use the `matchDepNames` option match each dep and add labels to them.

Btw this will work only for the cases where `depNameTemplate` is used.
You can check out the [packageRules ](https://docs.renovatebot.com/configuration-options/#packagerules) section to know more about other matching options.

```json
"packageRules": [
    {
      "matchDepNames": ["org.jenkins-ci.tests:plugins-compat-tester-cli"],
      "labels": ["dependencies", "full-test"]
    },
     {
      "matchDepNames": ["org.jenkins-ci.main:jenkins-war"],
      "labels": ["dependencies", "weekly-test"]
    }
  ]
```

**Reply by basil** [reply]
Thank you very much. I have implemented this successfully in https://github.com/jenkinsci/bom/pull/2648.
